ssg/utils.py

```diff
@@ -252,15 +252,28 @@ def mkdir_p(path):
             raise
 
 
-def banner_regexify(banner_text):
+def escape_regex(text):
     # We could use re.escape(), but it escapes too many characters, including plain white space.
     # In python 3.7 the set of charaters escaped by re.escape is reasonable, so lets mimic it.
     # See https://docs.python.org/3/library/re.html#re.sub
     # '!', '"', '%', "'", ',', '/', ':', ';', '<', '=', '>', '@', and "`" are not escaped.
-    banner_text = re.sub(r"([#$&*+-.^`|~:()])", r"\\\1", banner_text)
-    banner_text = banner_text.replace("\n", "BFLMPSVZ")
-    banner_text = banner_text.replace(" ", "[\\s\\n]+")
-    return banner_text.replace("BFLMPSVZ", "(?:[\\n]+|(?:\\\\n)+)")
+    return re.sub(r"([#$&*+-.^`|~:()])", r"\\\1", text) \
+        .replace("\n", "BFLMPSVZ") \
+        .replace(" ", "[\\s\\n]+") \
+        .replace("BFLMPSVZ", "(?:[\\n]+|(?:\\\\n)+)")
```

Comment:
I think the last 3 replaces are specific for regexes for banners.
But the first replace is generic enough.

Reply:
Hmm, hmm. Okay, maybe newlines are not really useful outside banners. But if we are creating universal filter wouldn't it be preferable to handle all cases? It's not like we are striving for tight performance in the build process itself.

Reply:
> But if we are creating universal filter wouldn't it be preferable to handle all cases?

I don't understand what is the purpose of this universal filter.

`banner_regexify` is not just about escaping, it also handles banner related quirks:
* [80 column folding](https://github.com/ComplianceAsCode/content/pull/6100/files#diff-6779cec91d09e974134b1e7cd05e420fR262), that is done on banners in `/etc/issue`.
* and newline [syntax differences](https://github.com/ComplianceAsCode/content/pull/6100/files#diff-6779cec91d09e974134b1e7cd05e420fR263) between `/etc/issue` and `/etc/dconf/db/gdm.d/`.

So I strongly prefer to keep `banner_regexify` separate from a general `escape_regex` function.

>  It's not like we are striving for tight performance in the build process itself.

I'm more worried about maintenance and readability, these banners can get quite complex.
See https://github.com/ComplianceAsCode/content/pull/5050 for some context.

Reply:
I think that only the [first replace](https://github.com/ComplianceAsCode/content/pull/6100/commits/aa4f77c6a7c5e88c433c7fef9f82ebc2b4392604#diff-6779cec91d09e974134b1e7cd05e420fR260) is suitable for `escape_regex`.

Reply:
Refined.